### PR TITLE
Bullet Physics Extension

### PR DESCRIPTION
**Pull request by devshgraphicsprogramming**

### CODE REVIEW

**Comment by devshgraphicsprogramming** [comment]
@khom-khun I'd remove `CCollisionShape` and inline pretty much every class method that isn't `virtual`

File: examples_tests/17.SimpleBulletIntegration/CMakeLists.txt [hunk 1]
```diff
@@ -4,4 +4,12 @@ if(NOT RES)
 	message(FATAL_ERROR "common.cmake not found. Should be in {repo_root}/cmake directory")
 endif()
 
-irr_create_executable_project("" "")
\ No newline at end of file
+set(BULLET_COLLISION_SRC "" CACHE STRING "For example 17.SimpleBulletIntegration - Not required")
```
Comment (devshgraphicsprogramming) [inline examples_tests/17.SimpleBulletIntegration/CMakeLists.txt]: what happens if can't find Bullet on system? Can it build without?

(BTW there is some code in the top-level CMake that skips building with Vulkan if Vulkan SDK is not detected on the system, maybe you can steal that approach)

Comment (khomsidakhun) [reply examples_tests/17.SimpleBulletIntegration/CMakeLists.txt]: I'd assume it won't spew errors and you'd be able to do without but just to be safe I'll add some sanity checks if they leave it blank and not glob_recurse over a blank string

**Comment by devshgraphicsprogramming** [comment]
Its a nice PR, it be a shame if it never got merged.

File: examples_tests/17.SimpleBulletIntegration/CMakeLists.txt (hunk 1 above)
Comment (devshgraphicsprogramming) [reply examples_tests/17.SimpleBulletIntegration/CMakeLists.txt]: now your Bullet lib needs to come from the submodule

**Comment by devshgraphicsprogramming** [comment]
Now basically add an option to cmake `IRR_BUILD_BULLET` (just like `IRR_BUILD_CEGUI`) that controls the building of your bullet example and the bullet dependency

Maybe merge the @manhnt9 sRGB branch into yours to access that nice `irr::ext` library building macro?

**Comment by devshgraphicsprogramming** [comment]
So I guess we're ready to merge?

**Pull request merged**